src/categories/categories.service.ts

```diff
@@ -1,15 +1,59 @@
-import { Injectable } from '@nestjs/common';
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
 import { CreateCategoryDto } from './dto/create-category.dto';
 import { UpdateCategoryDto } from './dto/update-category.dto';
+import { Category } from '../entities/category.entity';
+import { Repository } from 'typeorm';
+import { ZaLaResponse } from 'src/common/helpers/response';
 
 @Injectable()
 export class CategoriesService {
-  create(createCategoryDto: CreateCategoryDto) {
-    return 'This action adds a new category';
+  constructor(
+    @InjectRepository(Category)
+    private readonly categoryRepo: Repository<Category>
+  ) {}
+
+  async createNewCategory(createCategoryDto: CreateCategoryDto):Promise<Category> {
+    try{
+      const category = await this.categoryRepo.findOne({
+        where: { name: createCategoryDto.name},
+      })
+
+      if(category) {
+        throw new BadRequestException(
+          ZaLaResponse.BadRequest(
+            'Existing Value',
+            `A category with name ${createCategoryDto.name} already exist`,
+            '400'
+          )
+        )
+      }
+
+      const newCategory = this.categoryRepo.create({
+        ...createCategoryDto,
+      });
+      const savedCategory = await this.categoryRepo.save(newCategory);
+      return savedCategory;
+    }
+    catch(err){
+      throw new BadRequestException(
+        ZaLaResponse.BadRequest(err.name, err.message, err.status)
+      )
+    }
   }
 
-  findAll() {
-    return `This action returns all categories`;
+  async findAllCategory(): Promise<Category[]> {
+    try{
+      const category = await this.categoryRepo.find({
+        select: { id: true, name: true, description: true}
```

Comment:
I don't understand what you did here, simply do `find()` and it returns all entries

Reply:
In the dummy service... they returned specific entries like the id, name and description

Reply:
Understood, what if the frontend need more than those fields? please remove the select filter